server/methods/core/orders.js

```diff
@@ -139,6 +139,18 @@ Meteor.methods({
       throw new Meteor.Error(403, "Access Denied");
     }
 
+    // Server-side check to make sure discount is not greater than orderTotal.
+    let orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
+
+    if (discount > orderTotal) {
+      const err = "Discount is greater than the cart total";
+      Logger.error(err);
+      throw new Meteor.Error("orders/approvePayment.discount-amount", err);
```

Comment:
@jshimko is this a standard we are using other places, or just something that was thought for this instance?

We definitely should come up with our own error reporting standard, as @jshimko and I were discussing, the current system with http status related codes doesn't make a lot of sense for some of the things we need it for.

I'd be happy to discuss further and take the lead on adding it into the docs somewhere if we don't already have something similar.